File: strategy-game-frontend/src/core/models/city.ts
import {BuildingType} from "./buildingType";
import {TileRef} from "./tileRef";
import {ProductionQueueEntry} from "./productionQueueEntry";

export interface City {
    cityId: string,
    name: string,
    size: number,
    growthProgress: number
    countryId: string,
    tile: TileRef,
    isProvinceCapital: boolean,
    buildings: ({
        type: BuildingType,
        tile: TileRef | null,
        active: boolean
    })[],
    productionQueue: ProductionQueueEntry[]
}

File: strategy-game-backend/src/main/kotlin/de/ruegnerlukas/strategygame/backend/common/models/City.kt
package de.ruegnerlukas.strategygame.backend.common.models

import de.ruegnerlukas.strategygame.backend.common.utils.RGBColor

class City(
    val cityId: String,
    val countryId: String,
    val tile: TileRef,
    val name: String,
    val color: RGBColor,
    val isProvinceCapital: Boolean,
    val buildings: MutableList<Building>,
    val productionQueue: MutableList<ProductionQueueEntry>,
    var size: Int,
    var growthProgress: Float,
    var popConsumedFood: Float = 0f,
    var popGrowthConsumedFood: Boolean = false
)